### ISSUE TITLE
Update pdf.js to version 4

### ISSUE THREAD

**Issue by lfoppiano** [lfoppiano/streamlit-pdf-viewer#36]
```
PDF.js 4.0 is a major release that contains a number of API changes, features and bugfixes. The following changes are highlighted:

    Conversion to JavaScript modules for the builds
    Conversion to Fluent for localization
    Optimizations for CPU and memory usage
    Improvements and bugfixes for the annotation editor and annotation rendering
    Removal of the deprecated SVG back-end
    Removal of other deprecated functionality and options

```

**Comment by lfoppiano** [comment]
One of the limitation of this release is that node should be >= 18.

**Comment by lfoppiano** [comment]
https://github.com/lfoppiano/streamlit-pdf-viewer/security/dependabot/1

**Comment by t29mato** [comment]
https://github.com/lfoppiano/streamlit-pdf-viewer/commit/5f09846fbc5b4cd71ab910f98b343d681e5808ed

<img width="743" alt="image" src="https://github.com/lfoppiano/streamlit-pdf-viewer/assets/30012556/e289a684-e8ae-46fe-a619-934a397c4bbf">
I've modified the method for loading pdfjs-dist and fixed transpilation settings.
With these changes, the text is now displayed on top of the PDF in my environment as shown in the attached file.
The position is still misaligned, though. 🤔 

Since I have updated several library versions, you will need to delete node_modules and package-lock.json once to run in your environment.
